demos/immersed_fem/immersed_fem.py.rst

```diff
@@ -0,0 +1,115 @@
+Creating Firedrake-compatible meshes in Gmsh
+============================================
+
+The purpose of this demo is to summarize the
+key structure of a ``gmsh.geo`` file that creates a
+Firedrake-compatible mesh. For more details about Gmsh, please
+refer to the `Gmsh documentation <http://gmsh.info/#Documentation>`_.
+
+The first thing we define are four corners of a rectangle.
+We specify the x,y, and z(=0) coordinates, as well as the target
+element size at these corners (which we set to 0.5). ::
+
+  Point(1) = {-6,  2, 0, 0.5};
+  Point(2) = {-6, -2, 0, 0.5};
+  Point(3) = { 6, -2, 0, 0.5};
+  Point(4) = { 6,  2, 0, 0.5};
+
+Then, we define 5 points to describe a circle. ::
+
+  Point(5) = {   0,    0, 0, 0.1};
+  Point(6) = { 0.5,    0, 0, 0.1};
+  Point(7) = {-0.5,    0, 0, 0.1};
+  Point(8) = {   0,  0.5, 0, 0.1};
+  Point(9) = {   0, -0.5, 0, 0.1};
+
+Then, we create 8 edges: 4 for the rectangle and 4 for the circle.
+Recall that the Gmsh command ``Circle`` requires the arc to be
+strictly smaller than :math:`\pi`. ::
+
+  Line(1) = {1, 4};
+  Line(2) = {4, 3};
+  Line(3) = {3, 2};
+  Line(4) = {2, 1};
+  Circle(5) = {8, 5, 6};
+  Circle(6) = {6, 5, 9};
+  Circle(7) = {9, 5, 7};
+  Circle(8) = {7, 5, 8};
+
+Then, we glue together the rectangle edges and, separately, the circle edges.
+Note that ``Line``, ``Circle``, and ``Curve Loop`` (as well as ``Physical Curve`` below)
+are all curves in Gmsh and must possess a unique tag. ::
+
+  Curve Loop( 9) = {1, 2, 3, 4};
+  Curve Loop(10) = {8, 5, 6, 7};
```

Comment:
I note that this syntax requires gmsh 4?

Reply:
yes, I added a sentence about it (line 8)